### PR TITLE
test: add rust runtime tests

### LINKED ISSUE #1295: Add runtime testing for Rust
#### Reason/Context
In https://github.com/asyncapi/modelina/pull/1272 we added the initial runtime setup for Java, now it's time to replicate it for Rust.

The reason we add these is to ensure that the code that is generated is semantically and syntactically correct, and the only way to achieve this is by testing the models at runtime.

This initial setup should be as simple as possible, just a Rust project, with a single test and a single generated model, and that's it.

Here is the general task list:

- [ ] Create a Rust project under; `test/runtime/runtime-rust`
- [ ] Add a simple test code generation script here: `test/runtime/runtime-rust.ts`. Check out the Java variant as an example https://github.com/asyncapi/modelina/blob/master/test/runtime/runtime-java.ts
- [ ]  Add the generation script to the package file: https://github.com/asyncapi/modelina/blob/ba0bd41e9803394f56872c5ec92f8e335a4d687a/package.json#L98
- [ ] Run the generation script
- [ ] Manually add a simple test for the generated models, could be as simple as just creating an instance of it.
- [ ] Add a simple Jest script, which just execute the Rust test: https://github.com/asyncapi/modelina/blob/master/test/runtime/runtime-java.spec.ts
- [ ] Add a simple script that will run the test: https://github.com/asyncapi/modelina/blob/ba0bd41e9803394f56872c5ec92f8e335a4d687a/package.json#L97
- [ ] Create a CI workflow that runs the runtime tests: https://github.com/asyncapi/modelina/blob/master/.github/workflows/runtime-java-testing.yml

And that's it 🎉

### PR DESCRIPTION
**Pull request by Shurtu-gal**
**Description**
Add Runtime testing for rust

**Related issue(s)**
Fixes #1295

### CODE REVIEW

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
It's probably a bug with th generated rust code 🙂 We had it for Java as well 🙂 See https://github.com/asyncapi/modelina/issues/1294. 

[The test input](https://github.com/asyncapi/modelina/blob/master/test/runtime/generic-input.json) contains all the different variants of models. And it might be the generator does not correctly handle them 100%. So I would create a bug and then ignore the errors. We can merge this PR regardless 🙂 

Remember to add the GitHub workflow as well 🙂

**Comment by Shurtu-gal** [comment]
Sure I will do it.

**Comment by jonaslagoni** [comment]
LGTM 🔥 👍

@leigh-johnson wanna have a look?

File: test/runtime/runtime-rust.spec.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import path from 'path';
+import { execCommand } from '../blackbox/utils/Utils';
+
+jest.setTimeout(100000);
```
[the remark this answers is not in the document]
Comment (Shurtu-gal) [reply test/runtime/runtime-rust.spec.ts]: Done 👍🏻.
The errors are due to error in generator (PartialEQ especially) right now.

**Comment by Shurtu-gal** [comment]
@jonaslagoni the required changes are done 👍🏻.
The errors are due to error in generator (PartialEQ especially) right now.

**Comment by jonaslagoni** [comment]
> The errors are due to error in generator (PartialEQ especially) right now. 

And that's perfect 😄 Thats exactly why we do these tests. Do you mind creating an issue for the bug (unless there already is an issue for it)?

**Comment by Shurtu-gal** [comment]
@jonaslagoni Did it. Here's the link https://github.com/asyncapi/modelina/issues/1361

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍

**Comment by jonaslagoni** [comment]
Awesome work @Shurtu-gal, feel free to expand on the Rust runtime testing if you wish, this is something that really catches a lot of good stuff as you have already seen 🔥

**Comment by jonaslagoni** [comment]
@all-contributors please add @Shurtu-gal for test and infra

**Pull request merged**